### ISSUE TITLE
OBOFormat parser: xrefs create rdfs:label annotations

### ISSUE THREAD

**Issue by leechuck** [owlcs/owlapi#415]
For some ontologies in OBO format, xref clauses in the OBO file result in rdfs:label annotations to the class, with a part of the xref as value.

Example code (using 4.1.0-RC2, but also present in earlier 4.X versions):

```
@Grapes([
	  @Grab(group='org.semanticweb.elk', module='elk-owlapi', version='0.4.2'),
	  @Grab(group='net.sourceforge.owlapi', module='owlapi-api', version='4.1.0-RC2'),
	  @Grab(group='net.sourceforge.owlapi', module='owlapi-apibinding', version='4.1.0-RC2'),
	  @Grab(group='net.sourceforge.owlapi', module='owlapi-impl', version='4.1.0-RC2')
	])


import org.semanticweb.elk.owlapi.ElkReasonerFactory;
import org.semanticweb.owlapi.apibinding.OWLManager;
import org.semanticweb.owlapi.reasoner.*
import org.semanticweb.owlapi.vocab.OWLRDFVocabulary;
import org.semanticweb.owlapi.model.*;
import org.semanticweb.owlapi.io.*;
import org.semanticweb.owlapi.owllink.*;
import org.semanticweb.owlapi.util.*;
import org.semanticweb.owlapi.search.*;

OWLOntologyManager oManager;
OWLDataFactory df = OWLManager.getOWLDataFactory() ;
OWLOntologyManager lManager = OWLManager.createOWLOntologyManager()
OWLOntologyLoaderConfiguration config = new OWLOntologyLoaderConfiguration()
config.setFollowRedirects(true)
config = config.setMissingImportHandlingStrategy(MissingImportHandlingStrategy.SILENT)
def fSource = new FileDocumentSource(new File('chebi.obo'))
def ontology = lManager.loadOntologyFromOntologyDocument(fSource, config)

def cl = df.getOWLClass(IRI.create("http://purl.obolibrary.org/obo/CHEBI_82978"))
EntitySearcher.getAnnotations(cl, ontology, df.getRDFSLabel()).each { annotation -> // OWLAnnotation
  println annotation
}
```

The desired output is:

`Annotation(rdfs:label "paliperidone"^^xsd:string)`

The output I receive is:
```
Annotation(rdfs:label "CAS Registry Number"^^xsd:string)
Annotation(rdfs:label "PubMed citation"^^xsd:string)
Annotation(rdfs:label "Reaxys Registry Number"^^xsd:string)
Annotation(rdfs:label "paliperidone"^^xsd:string)
Annotation(rdfs:label "HMDB"^^xsd:string)
Annotation(rdfs:label "Wikipedia"^^xsd:string)
Annotation(rdfs:label "DrugBank"^^xsd:string)
Annotation(rdfs:label "KEGG DRUG"^^xsd:string)
```

The annotations are generated from the following statements in the OBO file:
```
xref: CiteXplore:25147315 "PubMed citation"
xref: CiteXplore:24962437 "PubMed citation"
xref: CiteXplore:24597755 "PubMed citation"
xref: KEGG DRUG:144598-75-4 "CAS Registry Number"
xref: CiteXplore:24289141 "PubMed citation"
xref: CiteXplore:24324228 "PubMed citation"
xref: CiteXplore:25377151 "PubMed citation"
xref: CiteXplore:24690136 "PubMed citation"
xref: KEGG DRUG:D05339 "KEGG DRUG"
xref: CiteXplore:24556260 "PubMed citation"
xref: CiteXplore:24462396 "PubMed citation"
xref: CiteXplore:24400982 "PubMed citation"
xref: Reaxys:8808385 "Reaxys Registry Number"
xref: CiteXplore:24829608 "PubMed citation"
xref: CiteXplore:25085446 "PubMed citation"
xref: CiteXplore:24491033 "PubMed citation"
xref: DrugBank:DB01267 "DrugBank"
xref: CiteXplore:24346811 "PubMed citation"
xref: Wikipedia:Paliperidone "Wikipedia"
xref: CiteXplore:23428785 "PubMed citation"
xref: HMDB:HMDB15396 "HMDB"
xref: CiteXplore:24752928 "PubMed citation"
xref: ChemIDplus:144598-75-4 "CAS Registry Number"
```
i.e., the rdfs:label is created from the source, in quotes, after the xrefs.

The problem seems to be in line 1392 of OWLAPIObo2Owl.java (https://github.com/owlcs/owlapi/blob/b2401a615a1151244a48b5c0ce88a5a98995a678/oboformat/src/main/java/org/obolibrary/obo2owl/OWLAPIObo2Owl.java#L1392) where an rdfs:label is created based on xref values. This breaks identifying the label for quite a large number of ontologies in OBO format.

**Comment by ignazio1977** [comment]
Do I understand it correctly, that the output you receive contains more annotations than expected? I.e., only one should be returned but you're getting eight?

**Comment by ignazio1977** [comment]
See https://github.com/owlcollab/oboformat/issues/89 for discussion

**Comment by leechuck** [comment]
Yes, there are too many annotations returned. It should return only the rdfs:label annotations to http://purl.obolibrary.org/obo/CHEBI_82978 but it returns (additionally) the rdfs:label annotations of the annotation axioms for http://purl.obolibrary.org/obo/CHEBI_82978. These should, as far as I understand the EntitySearcher, not be returned.

**Comment by ignazio1977** [comment]
OK, thank you. It is possible that entitysearcher is actually trying to be helpful, retrieving annotations from axioms where the subject is the entity - that's often useful, but not in this case.
I'll review the method and see if it needs changing.

**Comment by ignazio1977** [comment]
I've checked with 3.5.2, and the behaviour when looking for annotations only on the entity is as expected.  (3) is implemented in both 3.5.2 and 4.x, the issue here is that EntitySearcher is selecting the annotation assertion axioms for the entity and collecting both the annotation that is the object of each axiom and the annotations /on/ the axiom itself. This is an useful behavior in some cases but not all, as in this case.
I'll add a separate method to collect only the annotation object of the assertion axioms.

**Comment by ignazio1977** [comment]
To be updated in version 5 as well
